docs/examples/201/vm-new-or-existing-conditions/main.bicep

```diff
@@ -0,0 +1,272 @@
+  allowed: [
+    'password'
+    'sshPublicKey'
+  ]
+  metadata: {
+    description: 'Type of authentication to use on the Virtual Machine.'
+  }
+  default: 'sshPublicKey'
+}
+param adminPasswordOrKey string {
+  metadata: {
+    description: 'Password or ssh key for the Virtual Machine.'
+  }
+  secure: true
+}
+param vmSize string {
+  metadata: {
+    description: 'Size for the Virtual Machine.'
+  }
+  default: 'Standard_A2_v2'
+}
+param storageNewOrExisting string {
+  metadata: {
+    description: 'Determines whether or not a new storage account should be provisioned.'
+  }
+  default: 'new'
+}
+param storageAccountName string {
+  metadata: {
+    description: 'Name of the storage account'
+  }
+  default: 'storage${uniqueString(resourceGroup().id)}'
+}
+param storageAccountType string {
+  metadata: {
+    description: 'Storage account type'
+  }
+  default: 'Standard_LRS'
+}
+param storageAccountResourceGroupName string {
+  metadata: {
+    description: 'Name of the resource group for the existing storage account'
+  }
+  default: resourceGroup().name
+}
+param virtualNetworkNewOrExisting string {
+  metadata: {
+    description: 'Determines whether or not a new virtual network should be provisioned.'
+  }
+  default: 'new'
+}
+param virtualNetworkName string {
+  metadata: {
+    description: 'Name of the virtual network'
+  }
+  default: 'VirtualNetwork'
+}
+param addressPrefixes array {
+  metadata: {
+    description: 'Address prefix of the virtual network'
+  }
+  default: [
+    '10.0.0.0/16'
+  ]
+}
+param subnetName string {
+  metadata: {
+    description: 'Name of the subnet'
+  }
+  default: 'default'
+}
+param subnetPrefix string {
+  metadata: {
+    description: 'Subnet prefix of the virtual network'
+  }
+  default: '10.0.0.0/24'
+}
+param virtualNetworkResourceGroupName string {
+  metadata: {
+    description: 'Name of the resource group for the existing virtual network'
+  }
+  default: resourceGroup().name
+}
+param publicIpNewOrExisting string {
+  metadata: {
+    description: 'Determines whether or not a new public ip should be provisioned.'
+  }
+  default: 'new'
+}
+param publicIpName string {
+  metadata: {
+    description: 'Name of the public ip address'
+  }
+  default: 'PublicIp'
+}
+param publicIpDns string {
+  metadata: {
+    description: 'DNS of the public ip address for the VM'
+  }
+  default: 'linux-vm-${uniqueString(resourceGroup().id)}'
+}
+param publicIpResourceGroupName string {
+  metadata: {
+    description: 'Name of the resource group for the public ip address'
+  }
+  default: resourceGroup().name
+}
+
+var nicName = '${vmName}-nic'
+var linuxConfiguration = {
+  disablePasswordAuthentication: true
+  ssh: {
+    publicKeys: [
+      {
+        path: '/home/${adminUsername}/.ssh/authorized_keys'
+        keyData: adminPasswordOrKey
+      }
+    ]
+  }
+}
+var publicIpAddressId = {
+  id: resourceId(publicIpResourceGroupName, 'Microsoft.Network/publicIPAddresses', publicIpName)
```

Comment:
Is this going to work? The publicIp resource could potentially be deployed to a different scope

Reply:
I'd go with:
```bicep
var publicIpRef = {
  id: publicIp.id
}
```

And probably just remove the concept of deploying these resources to different rg scopes, since we don't support that in Bicep without modules